rules/java-net/src/main/resources/otagent/java-net.btm

```diff
@@ -0,0 +1,110 @@
+
+RULE java-net: Java HttpUrlConnection connect entry
+CLASS ^java.net.HttpURLConnection
+METHOD connect()
+HELPER io.opentracing.contrib.agent.OpenTracingHelper
+AT ENTRY
+IF !$0.connected && retrieveSpan($0) == null && !ignore($0)
+DO
+  associateSpan($0, getTracer().buildSpan($0.getRequestMethod())
+    .withTag(io.opentracing.tag.Tags.SPAN_KIND.getKey(), io.opentracing.tag.Tags.SPAN_KIND_CLIENT)
+    .withTag(io.opentracing.tag.Tags.HTTP_URL.getKey(), $0.getURL().toString())
+    .withTag("http.query", $0.getURL().getQuery())
+    .asChildOf(currentSpan())
+    .start());
+  getTracer().inject(retrieveSpan($0).context(), io.opentracing.propagation.Format$Builtin.HTTP_HEADERS,
+    new io.opentracing.contrib.agent.jnet.propagation.HttpURLConnectionInjectAdapter($0));
+ENDRULE
```

Comment:
I was looking for this... Why not move this into the actual Java code and just call a helper method from here? Makes it a lot easier to maintain & upgrade in case of OT API changes.